code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -440,3 +440,45 @@
 		M.emote(pick("twitch","laugh","frown"))
 	..()
 	. = 1
+
+/datum/reagent/drug/pumpup
+	name = "Pump-Up"
+	description = "Take on the world!"
+	reagent_state = LIQUID
+	color = "#e38e44"
+	overdose_threshold = 10
+
+/datum/reagent/drug/pumpup/on_mob_metabolize(mob/living/L)
+	..()
+	ADD_TRAIT(L, TRAIT_STUNRESISTANCE, type)
+
+/datum/reagent/drug/pumpup/on_mob_end_metabolize(mob/living/L)
+	REMOVE_TRAIT(L, TRAIT_STUNRESISTANCE, type)
+	..()
+
+/datum/reagent/drug/pumpup/on_mob_life(mob/living/carbon/M)
+	var/high_message = pick("Go! Go! GO!", "You feel ready...", "You feel invincible...")
+	M.Jitter(5)
+
+	if(prob(5))
+		to_chat(M, "<span class='notice'>[high_message]</span>")
+	if(prob(15))
+		M.losebreath++
+		M.adjustToxLoss(2, 0)
+	..()
+	. = 1
+
+/datum/reagent/drug/pumpup/overdose_start(mob/living/M)
+	to_chat(M, "<span class='userdanger'>You can't stop shaking, feeling your heart beat faster and faster!</span>")
+
+/datum/reagent/drug/pumpup/overdose_process(mob/living/M)
+	M.Jitter(5)
+	if(prob(5))
+		M.drop_all_held_items()
+	if(prob(20))
+		M.emote(pick("twitch","drool"))
+	if(prob(35))
```

Comment:
this is a nasty OD from just the O2 perspective (It does Tox damage so hard to use perf for this guy)

Reply:
Opinion on an alternative? I don't know the new chems so not sure the balance on damage.

Reply:
prob(35) is high

Reply:
perf deals 0.5 min tox damage OR scales with amount of O2 healed (5 oxy heal = 2 tox damage) if that helps.

Reply:
It should be noted that the OD threshold is low enough that you can OD someone on this chem by shooting them with a syringe of it.

Reply:
Yeah I’m trying to figure out the best thing to do here. I think it might be higher od, you get more of the chem when you make it, but it metabolises much faster.